### ISSUE TITLE
Pressure solver performance for PNPN

### ISSUE THREAD

**Issue by stgeke** [Nek5000/Nek5000#96]
The pressure solve with PN/PN-2 is 1.5x faster e.g. for turbChannel. Same applies for other cases. It looks you'll get less pressure iterations with PN/PN-2 compared to PN/PN. Not sure why - I guess the implementation of preconditioned is somehow different. Something we need to look at.

**Comment by stgeke** [comment]
@ggiannako  If I recall correctly, you did performance experiments. Can you add more details?

**Comment by ggiannako** [comment]
In a small 3D engine sim case (E=75k, lx1=6), for 100 steps (fixed dt), the average number of pressure iterations for PN/PN is 36% higher than PN/PN-2. In this case, this results to about 23% higher runtime.

**Comment by fischer1** [comment]
Can you send me the logfiles ?

Paul

________________________________

**Comment by rvinuesa** [comment]
Hi,

I have some data from a channel flow comparison we did quite some time ago, so not sure how relevant it would be. It is from 2014, and was done with version 758. We have three relevant cases:

1) 97,344 elements, N=11, Pn-Pn-2. Time per time step: 8.05e-1

2) 97,344 elements, N=11, Pn-Pn. Time per time step: 8.18e-1 [around 2% higher]

3) 97,344 elements, N=11, Pn-Pn + characteristics. Time per time step: 2.74, for an increase in time step of a factor of 2. So if one would multiply the value 8.18e-1 by 2 one would get 1.63, still much lower than the value with characteristics.

I think that at the time we concluded that the performance was similar between Pn-Pn-2 and Pn-Pn (perhaps in some runs the difference was a bit more than 2%, but not so much).

Best regards,

Ricardo.
